src/db.c

```diff
@@ -835,7 +835,7 @@ int parseScanCursorOrReply(client *c, robj *o, unsigned long *cursor) {
      * getLongLongFromObject() does not cover the whole cursor space. */
     errno = 0;
     *cursor = strtoul(o->ptr, &eptr, 10);
-    if (isspace(((char*)o->ptr)[0]) || eptr[0] != '\0' || errno == ERANGE)
+    if (sdslen(o->ptr) == 0 || isspace(((char*)o->ptr)[0]) || eptr[0] != '\0' || errno == ERANGE)
```

Comment:
Should we instead just check that first char `!isdigit` instead of `isspace`?

Reply:
What other, similar, places did you have in mind, maybe I just don't have right context?
I don't feel strongly about this, but the form I've proposed above seems to be most concise, yet readable way to express what we want. Cursor is unsigned, so it should be fine to check if first char is a digit.

Reply:
i mean all the other places that call this family of APIs (strtoll, strtod) and use `&eptr` to validate they parsed the whole string.
e.g. there's one in setConfigOOMScoreAdjValuesOption, one in sortCommandGeneric, and many others.

Reply:
@oranagra Hello, I would like you to just help me merge this PR, and I will continue to pay attention to all other places susceptible to this error in the future, and I will pay attention to those unexpected modifications (such as Spaces) in the future. Thank you again for helping me merge this PR, and I will definitely continue to pay attention to the follow-up problems of redis

Reply:
i think i'd rather look at the wider problem in one go, rather than solving the issues one by one (which could lead to some realizations and then repeated changes to the same code).
and either way, i can't merge this PR in it's current form since it contains quite a few unintended changes.